src/tests/ftest/deployment/critical_integration.py

```diff
@@ -97,13 +99,36 @@ def test_ras(self):
         dmg = self.get_dmg_command()
         rank_list = self.server_managers[0].get_host_ranks(self.hostlist_servers)
         self.log.info("rank_list: %s", rank_list)
-        for rank in rank_list:
-            dmg.system_stop(ranks=rank)
-            if self.server_managers[0].check_rank_state(rank, "stopped", 5):
-                dmg.system_start(ranks=rank)
-                if not self.server_managers[0].check_rank_state(rank, "joined", 5):
-                    self.fail("Rank {} failed to restart".format(rank))
-            else:
-                self.fail("Rank {} failed to stop".format(rank))
+        half_num_ranks = len(rank_list)//2
+        # divide total ranks list into two halves to save time during system stop
+        sub_rank_list = [rank_list[x:x+half_num_ranks] for x in range(0, len(rank_list),
+                                                                      half_num_ranks)]
+        self.log.info("sub_rank_list: %s", sub_rank_list)
+
+        # stop ranks, verify they stopped successfully and restart the stopped ranks
+        since = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
+        for sub_list in sub_rank_list:
+            ranks_to_stop = ",".join([str(rank) for rank in sub_list])
```

Comment:
Why stop half the ranks at a time instead of just stopping all of them?

Reply:
I thought having a scenario where there are some active connections vs some are stopped will be somewhat closer to real world scenario. Though it doesn't matter much in this test case if all are on or off.